func.go

```diff
@@ -130,6 +161,23 @@ func WithEnv(envs map[string]string) FuncExecOption {
 	}
 }
 
+// WithHandleFunc adds a handle func that will handle all commands
+func WithHandleFunc(fn CmdFunc) FuncExecOption {
+	return func(fExec *FuncExec) {
+		if fExec.mux == nil {
+			fExec.mux = NewMux()
+		}
+		fExec.mux.HandleFunc(pat("*"), fn)
+	}
+}
+
+// WithMux sets a command Mux to the funcExec
+func WithMux(mux *Mux) FuncExecOption {
```

Comment:
Why have these as functional options? Are these really optional? Now that we have a unified `mux` type that we should require the FuncExec always gets a mux. We can expose simple muxes like `Success` or `Fail` as well as supporting the function map or a more robust mux.